hotwax-gift-wrapper/extensions/gift-wrap/src/BlockExtension.jsx

```diff
@@ -0,0 +1,319 @@
+import {
+  reactExtension,
+  useApi,
+  AdminBlock,
+  BlockStack,
+  Text,
+  Select,
+  Button,
+  InlineStack,
+  ChoiceList,
+  TextArea,
+} from "@shopify/ui-extensions-react/admin";
+import { Banner, Heading } from "@shopify/ui-extensions/admin";
+import { useEffect, useState } from "react";
+import {
+  getDraftOrderItems,
+  updateDraftOrder,
+  removeCustomAtrribute,
+} from "../utils/shopifyUpdateHandler";
+
+const TARGET = "admin.draft-order-details.block.render";
+
+export default reactExtension(TARGET, () => <App />);
+
+function App() {
+  const { data, query, navigation } = useApi(TARGET);
+  const [items, setItems] = useState([]);
+  const [selectedItemIds, setSelectedItemIds] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [giftWrapOption, setGiftWrapOption] = useState(
+    "Gift Wrap Items Together"
+  );
+  const [giftWrapForWrappingTogether, setGiftWrapForWrappingTogether] =
+    useState("With Love");
+  const [giftNote, setGiftNote] = useState("");
+  const [updateMessage, setUpdateMessage] = useState("");
+
+  const giftWrapOptions = [
+    { value: "Gift Wrap Items Together", label: "Together" },
+    { value: "Gift Wrap Items Separately", label: "Separate" },
+  ];
+
+  const wrapperOptions = [
+    { value: "Holiday", label: "Holiday" },
+    { value: "With Love", label: "With Love" },
+  ];
+
+  const draftOrderId = data?.selected[0]?.id;
+
+  useEffect(() => {
+    fetchDraftOrderItems();
+  }, [data]);
+
+  async function fetchDraftOrderItems() {
+    if (draftOrderId) {
+      try {
+        const response = await getDraftOrderItems(draftOrderId);
+        const draftOrder = response?.data?.draftOrder;
+        
+        if (draftOrder) {
+          setGiftNote(draftOrder.note || "");
+          setGiftWrapOption(
+            draftOrder.customAttributes?.find(
+              (attr) => attr.key === "Gift Wrap Option"
+            )?.value || "Gift Wrap Items Together"
+          );
+
+          if (giftWrapOption === "Gift Wrap Items Together") {
+            const wrapValue =
+              draftOrder.lineItems.edges
+                .find(({ node }) =>
+                  node.customAttributes?.some(
+                    (attr) => attr.key === "Gift Wrap"
+                  )
+                )
+                ?.node?.customAttributes?.find(
+                  (attr) => attr.key === "Gift Wrap"
+                )?.value || "With Love";
```

Comment:
This logic can be simplified as we have iterating the attributes twice but we can achieve the same using a single iteration.